src/main/java/uk/nhs/digital/nhsconnect/nhais/service/SequenceService.java

```diff
@@ -0,0 +1,43 @@
+package uk.nhs.digital.nhsconnect.nhais.service;
+
+import io.micrometer.core.instrument.util.StringUtils;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.nhs.digital.nhsconnect.nhais.exceptions.SenderAndRecipientValidationException;
+import uk.nhs.digital.nhsconnect.nhais.repository.SequenceRepository;
+
+@Service
+@AllArgsConstructor
+public class SequenceService {
+    private final static String TRANSACTION_ID = "transaction_id";
+    private final static String INTERCHANGE_FORMAT = "SIS-%s-%s";
+    private final static String INTERCHANGE_MESSAGE_FORMAT = "SMS-%s-%s";
+
+    private SequenceRepository sequenceRepository;
+
+    public String generateTransactionId() {
```

Comment:
Also wondering if we should concern the actual numbers? They get leading zero-padded to (7?) digits and translation. Outbound state will need to store them as numbers for easier sorting.